iceoryx_examples/icediscovery_in_c/iox_c_find_service.c

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_binding_c/runtime.h"
+#include "iceoryx_binding_c/service_discovery.h"
+#include "sleep_for.h"
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#define SEARCH_RESULT_CAPACITY 10
+
+bool keepRunning = true;
+
+const char APP_NAME[] = "iox-c-find-service";
+
+static void sigHandler(int signalValue)
+{
+    // Ignore unused variable warning
+    (void)signalValue;
+    // caught SIGINT or SIGTERM, now exit gracefully
+    keepRunning = false;
+}
+
+void printSearchResult(const iox_service_description_t service)
+{
+    printf(
+        "- Service: %s, Instance: %s, Event: %s\n", service.serviceString, service.instanceString, service.eventString);
+}
+
+void searchFrontCameras(const iox_service_description_t service, void* count)
+{
+    if (strcmp(service.instanceString, "FrontLeft") == 0 || strcmp(service.instanceString, "FrontRight") == 0)
```

Comment:
There is this advice to use https://www.cplusplus.com/reference/cstring/strncmp/ to avoid errors due to missing zero termination but I am not sure we want to do this here (code gets uglier).  Up to you.

The literal is obviously zero-terminated, the strings in `service` basically are returned from our API so ...